### PR TITLE
[d3d9] Force D3DTADDRESS_CLAMP for cubes.

### PR DESCRIPTION
**Pull request by DadSchoorse**
matches wined3d

### CODE REVIEW

File: src/d3d9/d3d9_device.cpp [hunk 1]
```diff
@@ -3783,10 +3783,14 @@ namespace dxvk {
       const bool newDepth = newTexture->IsShadow();
 
       if (oldDepth != newDepth) {
-        m_depthTextures &= ~(1u << StateSampler);
-        if (newDepth)
-          m_depthTextures |= 1u << StateSampler;
+        m_depthTextures ^= 1u << StateSampler;
```
Comment (Joshua-Ashton) [inline src/d3d9/d3d9_device.cpp:3786]: Can we separate out xor change into another commit in case we mess something up here?

Comment (DadSchoorse) [reply src/d3d9/d3d9_device.cpp:3786]: done
